packages/shared/src/transactions.ts

```diff
@@ -151,32 +151,31 @@ export const isSentTx = (
   tx: e.Transaction | e.PendingTransaction | e.MempoolTransaction | SentTransaction
 ): tx is SentTransaction => 'status' in tx
 
-export const isInternalTx = (
+export const isWalletSelfTransfer = (
   tx: e.Transaction | e.PendingTransaction | e.MempoolTransaction,
-  internalAddresses: AddressHash[]
-): boolean =>
-  [...(tx.outputs ?? []), ...(tx.inputs ?? [])].every(
-    (io) => io?.address && internalAddresses.includes(getBaseAddressStr(io.address))
-  )
+  walletAddresses: AddressHash[]
+): boolean => getTxAddresses(tx).every((address) => address && walletAddresses.includes(getBaseAddressStr(address)))
 
 export const isContractTx = (tx: e.Transaction | e.PendingTransaction | e.MempoolTransaction): boolean =>
   !!tx.outputs?.some(inputOutputIsContractAddress) || !!tx.inputs?.some(inputOutputIsContractAddress)
 
 const inputOutputIsContractAddress = (io: e.Input | e.Output): boolean => !!io.address && isContractAddress(io.address)
 
+const getTxAddresses = (tx: e.Transaction | e.PendingTransaction | e.MempoolTransaction): (AddressHash | undefined)[] =>
+  [...(tx.outputs ?? []), ...(tx.inputs ?? [])].map((io) => io.address)
```

Comment:
Should we dedupe this list? Or maybe add a param to let us do it? 🤔 
I'm wondering, because the result of this function is often re-mapped, like above at line 34 and 157. 
In line 157, we actually map twice, so duplicated addresses probably have an impact on perfs.

Reply:
Discussed and resolved on slack.